<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.500747;0.499253">
     <DockArea name="...">
      <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
       <range bottom="-1.150140" top="1.154742" right="95.280033" left="81.338075"/>
       <limitY/>
       <curve name="/diana_gripper/joint_states/diana_gripper/LFJ1/position" color="#1f77b4"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/LFJ2/position" color="#d62728"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/RFJ1/position" color="#1ac938"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/RFJ2/position" color="#ff7f0e"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
       <range bottom="-0.968965" top="1.071230" right="95.276137" left="81.338075"/>
       <limitY/>
       <curve name="/diana_gripper/joint_states/diana_gripper/RFJ2/velocity" color="#f14cc1"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/RFJ1/velocity" color="#9467bd"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/LFJ2/velocity" color="#17becf"/>
       <curve name="/diana_gripper/joint_states/diana_gripper/LFJ1/velocity" color="#bcbd22"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.500747;0.499253">
     <DockArea name="...">
      <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
       <range bottom="-0.019687" top="-0.009341" right="9.993976" left="0.015940"/>
       <limitY/>
       <curve name="/diana_gripper/palm_imu/imu_raw/angular_velocity/x" color="#1f77b4"/>
       <curve name="/diana_gripper/palm_imu/imu_raw/angular_velocity/y" color="#d62728"/>
       <curve name="/diana_gripper/palm_imu/imu_raw/angular_velocity/z" color="#1ac938"/>
      </plot>
     </DockArea>
     <DockArea name="...">
      <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
       <range bottom="-0.208007" top="5.007627" right="9.993976" left="0.015940"/>
       <limitY/>
       <curve name="/diana_gripper/palm_imu/imu_raw/linear_acceleration/x" color="#ff7f0e"/>
       <curve name="/diana_gripper/palm_imu/imu_raw/linear_acceleration/y" color="#f14cc1"/>
       <curve name="/diana_gripper/palm_imu/imu_raw/linear_acceleration/z" color="#9467bd"/>
      </plot>
     </DockArea>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab3">
   <Container>
    <DockSplitter orientation="-" count="2" sizes="0.500747;0.499253">
     <DockSplitter orientation="|" count="2" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
        <range bottom="58.950000" top="61.050000" right="38.427981" left="28.523958"/>
        <limitY/>
        <curve name="/diana_gripper/voltages/axes.0" color="#17becf"/>
        <curve name="/diana_gripper/voltages/axes.1" color="#bcbd22"/>
        <curve name="/diana_gripper/voltages/axes.2" color="#1f77b4"/>
        <curve name="/diana_gripper/voltages/axes.3" color="#d62728"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
        <range bottom="-595.200000" top="24403.200000" right="38.430007" left="28.526004"/>
        <limitY/>
        <curve name="/diana_gripper/currents/axes.0" color="#17becf"/>
        <curve name="/diana_gripper/currents/axes.1" color="#bcbd22"/>
        <curve name="/diana_gripper/currents/axes.2" color="#1f77b4"/>
        <curve name="/diana_gripper/currents/axes.3" color="#d62728"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" count="2" sizes="0.5;0.5">
      <DockArea name="...">
       <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
        <range bottom="19.925000" top="23.075000" right="38.425987" left="28.521979"/>
        <limitY/>
        <curve name="/diana_gripper/temperatures/axes.0" color="#1ac938"/>
        <curve name="/diana_gripper/temperatures/axes.1" color="#ff7f0e"/>
        <curve name="/diana_gripper/temperatures/axes.2" color="#f14cc1"/>
        <curve name="/diana_gripper/temperatures/axes.3" color="#9467bd"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot flip_x="false" mode="TimeSeries" flip_y="false" style="Lines">
        <range bottom="-243.600000" top="243.600000" right="38.417991" left="28.434015"/>
        <limitY/>
        <curve name="/diana_gripper/torques/axes.0" color="#1ac938"/>
        <curve name="/diana_gripper/torques/axes.1" color="#ff7f0e"/>
        <curve name="/diana_gripper/torques/axes.2" color="#f14cc1"/>
        <curve name="/diana_gripper/torques/axes.3" color="#9467bd"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default delimiter="0" time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;]]--&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. ]]--&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value  &#xa;&#xa;]]--&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

